daft/execution/rust_physical_plan_shim.py

```diff
@@ -26,41 +25,43 @@
 
 
 def scan_with_tasks(
-    scan_tasks: Iterator[ScanTask],
+    scan_task: ScanTask,
 ) -> physical_plan.InProgressPhysicalPlan[PartitionT]:
     """child_plan represents partitions with filenames.
 
     Yield a plan to read those filenames.
     """
-    # TODO(Clark): Bundle scan tasks into single-instruction bulk reads.
-    for scan_task in scan_tasks:
-        scan_task_batch = ScanTaskBatch.from_scan_tasks([scan_task])
+    for i in range(len(scan_task)):
+        # TODO(Clark): We currently hardcode this to have len-1-ScanTaskes per instruction.
+        # We can instead right-size and bundle the ScanTask into single-instruction bulk reads.
+        single_task_batch = scan_task.slice(i, i + 1)
```

Comment:
Why are slicing the ScanTask, the ScanTask be issued as a single instruction?

Reply:
Fixed and now the Python API expects a list[ScanTask] instead